pkg/controller/statusmanager/pod_status.go

```diff
@@ -98,77 +99,96 @@ func SetNodeCondition(conditions *[]corev1.NodeCondition, newCondition corev1.No
 // CheckExistingAgentPods is for a case: nsx-node-agent becomes unhealthy -> NetworkUnavailable=True -> operator off ->
 // nsx-node-agent becomes healthy and keeps running -> operator up -> operator cannot receive nsx-node-agent event to
 // set NetworkUnavailable=False. So a full sync at the start time is necessary.
-func (status *StatusManager) CheckExistingAgentPods(firstBoot *bool, sharedInfo *sharedinfo.SharedInfo) error {
+func (status *StatusManager) CheckExistingAgentPods (firstBoot *bool, sharedInfo *sharedinfo.SharedInfo) (reconcile.Result, error) {
 	status.Lock()
 	defer status.Unlock()
 
 	if !*firstBoot {
-		return nil
+		return reconcile.Result{}, nil
 	}
 	log.Info("Checking all nsx-node-agent pods for node condition")
 	pods := &corev1.PodList{}
 	err := status.client.List(context.TODO(), pods, client.MatchingLabels{"component": operatortypes.NsxNodeAgentContainerName})
 	if err != nil {
 		log.Error(err, "Error getting pods for node condition")
-		return err
+		return reconcile.Result{}, err
 	}
 	if len(pods.Items) == 0 {
 		log.Info("nsx-node-agent not found for node condition")
 		nodes := corev1.NodeList{}
 		err = status.client.List(context.TODO(), &nodes)
 		if err != nil {
 			log.Error(err, "Failed to get nodes for condition updating")
-			return err
+			return reconcile.Result{}, err
 		}
 		for _, node := range nodes.Items {
-			status.setNodeNetworkUnavailable(node.ObjectMeta.Name, false, nil, "NSXNodeAgent", "Waiting for nsx-node-agent to be created", sharedInfo)
+			// In this case the nsx-node-agent is not created yet so we don't need to care about the graceful time
+			_, err := status.setNodeNetworkUnavailable(node.ObjectMeta.Name, false, nil, "NSXNodeAgent", "Waiting for nsx-node-agent to be created", sharedInfo)
+			if err != nil {
+				return reconcile.Result{}, err
+			}
 		}
 		*firstBoot = false
-		return nil
+		return reconcile.Result{}, nil
 	}
 	podName := types.NamespacedName{}
 	for _, pod := range pods.Items {
-		status.setNodeConditionFromPod(podName, sharedInfo, &pod)
+		if _, err := status.setNodeConditionFromPod(podName, sharedInfo, &pod); err != nil {
+			return reconcile.Result{}, err
+		}
 	}
 	*firstBoot = false
-	return nil
+	return reconcile.Result{}, nil
 }
 
-func (status *StatusManager) setNodeNetworkUnavailable(nodeName string, ready bool, startedAt *time.Time, reason string, message string, sharedInfo *sharedinfo.SharedInfo) {
-	if sharedInfo.LastNetworkAvailable == nil {
-		sharedInfo.LastNetworkAvailable = make(map[string]time.Time)
+// In general, write your controller with the assumption that information will eventually be correct, but may be slightly out of date.
```

Comment:
Please start with function name for comments

Reply:
Done